conformance/tests/generics_basic.py

```diff
@@ -0,0 +1,154 @@
+"""
+Tests for basic usage of generics.
+"""
+
+# Specification: https://typing.readthedocs.io/en/latest/spec/generics.html#introduction
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Any, Generic, TypeVar, assert_type
+
+T = TypeVar('T')
+
+# > Generics can be parameterized by using a factory available in
+# > ``typing`` called ``TypeVar``.
+
+def first(l: Sequence[T]) -> T:
+    return l[0]
+
+
+def test_first(seq_int: Sequence[int], seq_str: Sequence[str]) -> None:
+    assert_type(first(seq_int), int)
+    assert_type(first(seq_str), str)
+
+# > ``TypeVar`` supports constraining parametric types to a fixed set of
+# > possible types
+
+AnyStr = TypeVar('AnyStr', str, bytes)
+
+def concat(x: AnyStr, y: AnyStr) -> AnyStr:
+    return x + y
+
+def test_concat(s: str, b: bytes, a: Any) -> None:
+    concat(s, s)  # OK
+    concat(b, b)  # OK
+    concat(s, b)  # Type error
+    concat(b, s)  # Type error
+
+    concat(s, a)  # OK
+    concat(a, b)  # OK
+
+# > Specifying a single constraint is disallowed.
+
+BadConstraint1 = TypeVar('BadConstraint', str)  # Type error
```

Comment:
I recommend changing the quoted name (the first argument) to match the assigned name "BadConstraint1" to avoid an unnecessary (and unrelated) error here.